src/metricscollector/collector/collector.go

```diff
@@ -17,64 +18,64 @@ type NoaaConsumer interface {
 }
 
 type Collector struct {
-	logger   lager.Logger
 	conf     *config.CollectorConfig
+	logger   lager.Logger
 	cfc      cf.CfClient
 	noaa     NoaaConsumer
 	database db.DB
-	tick     *time.Ticker
+	cclock   clock.Clock
 	doneChan chan bool
 	pollers  map[string]*AppPoller
+	ticker   clock.Ticker
 	lock     *sync.Mutex
 }
 
-func NewCollector(logger lager.Logger, conf *config.CollectorConfig, cfc cf.CfClient, noaa NoaaConsumer, database db.DB) *Collector {
+func NewCollector(conf *config.CollectorConfig, logger lager.Logger, cfc cf.CfClient, noaa NoaaConsumer, database db.DB, cclock clock.Clock) *Collector {
 	return &Collector{
-		logger:   logger,
 		conf:     conf,
+		logger:   logger,
 		cfc:      cfc,
 		noaa:     noaa,
 		database: database,
+		cclock:   cclock,
 		doneChan: make(chan bool),
 		pollers:  make(map[string]*AppPoller),
 		lock:     &sync.Mutex{},
 	}
 }
 
 func (c *Collector) Start() {
-	c.tick = time.NewTicker(time.Duration(c.conf.RefreshInterval) * time.Second)
+	c.ticker = c.cclock.NewTicker(time.Duration(c.conf.RefreshInterval) * time.Second)
```

Comment:
The config should make the RefreshInterval a duration. We can expose it as a duration or number of seconds in BOSH later.